portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -86,12 +72,59 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         response.sendRedirect("/contact.html");
     }
 
+    /** Returns comment limit value from the request, or -1 if user selected 'all' */
+    private int getCommentLimit(HttpServletRequest request) {
+        String limit_string = request.getParameter("comment-limit");
+        int comment_limit = 0;
+
+        try {
+            comment_limit = Integer.parseInt(limit_string);
+        } catch (NumberFormatException n) {
+            if (limit_string.equals("all")) {
```

Comment:
If you wanted to streamline your backend code, you could use a -1 result from the html control. It is possible to set a different value for a menu control than the text. (By default it will be the same, but can be made different). See more discussion there.